pkg/cfn/manager/nodegroup.go

```diff
@@ -74,6 +74,45 @@ func (c *StackCollection) createManagedNodeGroupTask(errorCh chan error, ng *api
 	return c.CreateStack(name, stack, ng.Tags, nil, errorCh)
 }
 
+func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(errorCh chan error, ng *api.ManagedNodeGroup) error {
+	// describe node group to retrieve ASG names
+	input := &eks.DescribeNodegroupInput{
+		ClusterName:   aws.String(c.spec.Metadata.Name),
+		NodegroupName: aws.String(ng.Name),
+	}
+	res, err := c.eksAPI.DescribeNodegroup(input)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't get managed nodegroup details for nodegroup %q", ng.Name)
+	}
+
+	// set the managed nodegroup tags to all the ASGs found
+	if res.Nodegroup.Resources != nil {
+		// build the input tags for all ASGs attached to the managed nodegroup
+		asgTags := []*autoscaling.Tag{}
+
+		for _, asg := range res.Nodegroup.Resources.AutoScalingGroups {
+			for ngTagKey, ngTagValue := range ng.Tags {
+				asgTag := &autoscaling.Tag{
+					ResourceId:        aws.String(*asg.Name),
+					ResourceType:      aws.String("auto-scaling-group"),
+					Key:               aws.String(ngTagKey),
+					Value:             aws.String(ngTagValue),
+					PropagateAtLaunch: aws.Bool(false),
+				}
+				asgTags = append(asgTags, asgTag)
+			}
+		}
+
+		input := &autoscaling.CreateOrUpdateTagsInput{Tags: asgTags}
+		if _, err := c.asgAPI.CreateOrUpdateTags(input); err != nil {
+			return errors.Wrapf(err, "creating or updating asg tags for managed nodegroup %q", ng.Name)
+		}
+	}
+
+	go func() { errorCh <- nil }()
```

Comment:
as with other tasks, this whole thing should be in a go routine with a defer close on the err channel.

Reply:
Try to take a look at `createClusterTask`.

Reply:
Alternatively, we do tasks now differently. :D But I'm not going to force that on this contribution, since it's not that trivial to implement out of the blue. :) But if you want to, things are under `pkg/actions` now-a-days.

Take a look at pkg/actions/karpenter for tasks and how they are initialised. If you wish. :)

Reply:
Done in 0099c487ccebe4d8141505c8db17fcaa3f158ff0
Tried to do it the "old way" for now :)